src/Moryx.Model/Annotations/DateTimeKindAnnotation.cs

```diff
@@ -23,13 +24,20 @@ public static class DateTimeKindAnnotation
         private const string AnnotationName = "DateTimeKind";
 
         private static readonly ValueConverter<DateTime, DateTime> UtcConverter =
-            new ValueConverter<DateTime, DateTime>(v => v, v => DateTime.SpecifyKind(v, DateTimeKind.Utc));
-
+            new ValueConverter<DateTime, DateTime>(v => ConvertToUtc(v), v => DateTime.SpecifyKind(v, DateTimeKind.Utc));
         private static readonly ValueConverter<DateTime, DateTime> LocalConverter =
-            new ValueConverter<DateTime, DateTime>(v => v, v => DateTime.SpecifyKind(v, DateTimeKind.Local));
-
+            new ValueConverter<DateTime, DateTime>(v => ConvertToUtc(v), v => DateTime.SpecifyKind(v, DateTimeKind.Local));
         private static readonly ValueConverter<DateTime, DateTime> UnspecifiedConverter =
-            new ValueConverter<DateTime, DateTime>(v => v, v => DateTime.SpecifyKind(v, DateTimeKind.Unspecified));
+           new ValueConverter<DateTime, DateTime>(v => ConvertToUtc(v), v => DateTime.SpecifyKind(v, DateTimeKind.Unspecified));
+
+        private static DateTime ConvertToUtc(DateTime dateTime)
```

Comment:
Why is everything converted to UTC? In my sample if provided an option to attribute your entity properties: More flexible for the developer, he can decide what is necessary